### PR TITLE
Update CertificateLoader.cs

### LINKED ISSUE #49062: HTTPS certificate exact match using Issuer's SimpleName instead of Certificate's
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have 2 certificates `NAME` and `NAMEB` (actual cert names changed for obvious reasons). I am trying to configure Kestrel to use the `NAMEB` cert but the `NAME` cert is always being chosen. These certs are not self-signed.

As part of #34582 the `Microsoft.AspNetCore.Server.Kestrel.Https.CertificateLoader` was changed to support exact subject name match but it appears the certificate's issuer SimpleName is being used instead of the certificate's SimpleName. The `certificate.GetNameInfo(X509NameType.SimpleName, true)` should be `certificate.GetNameInfo(X509NameType.SimpleName, false)`

It should be noted that the `NAME` cert expires after the `NAMEB` cert which is why it is being chosen as the the first found cert.

### Expected Behavior

I would expect the exact match logic to work and the `NAME` cert is chosen.

### Steps To Reproduce

Kestrel config:
```
"Kestrel": {
    "Endpoints": {
      "HttpsInlineCertStore": {
        "Url": "https://NAME:7286",
        "Certificate": {
          "Subject": "NAME",
          "Store": "MY",
          "Location": "LocalMachine"
        }
      }
    }
  }
```

With the following 2 certs in the cert store:
* `NAME`
    * Expires after `NAMEB` 
* `NAMEB`

### Exceptions (if any)

_No response_

### .NET Version

7.0.304

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by simon-curtis**
# CertificateLoader loading wrong certificate

Fixes certificate selection by subject name

## Description

The check for exact match to certificate subject fails as the `forIssuer` flag is set to true. This returns the `issuer` and not the `subject` this means that if two certificates have a partial match, it will just select the first one that has the search value in the name.

Fixes #49062 (in this specific format)

### CODE REVIEW

**Comment by martincostello** [comment]
I imagine the team would want a test to be added for this if the change is taken 😃

**Comment by simon-curtis** [comment]
I would tend to agree, but how do you provide a unit test for something to environment specific? I.e. there aren't any tests covering this now.

**Comment by amcasey** [comment]
> I would tend to agree, but how do you provide a unit test for something to environment specific? I.e. there aren't any tests covering this now.

It's possible we have a way to mock the store.  Let me see if I can find a test that does that (there may not be one).

File: src/Servers/Kestrel/Core/src/CertificateLoader.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static X509Certificate2 LoadFromStoreCert(string subject, string storeNam
                     // Pick the first one if there's no exact match as a fallback to substring default.
                     foundCertificate ??= certificate;
 
-                    if (certificate.GetNameInfo(X509NameType.SimpleName, true).Equals(subject, StringComparison.InvariantCultureIgnoreCase))
```
Comment (amcasey) [inline src/Servers/Kestrel/Core/src/CertificateLoader.cs:51]: While you're here, can you please change this to a named argument?

**Comment by amcasey** [comment]
This looks correct to me, but let's see if we can figure out a way to test this.

**Comment by amcasey** [comment]
Re-running the tests, since the failure is unrelated.
/azp run

**Comment by amcasey** [comment]
There's one test in HttpsConnectionMiddlewareTests, but it's not great.  Given that the original PR didn't add a test and the fix seems self-explanatory (after making it a named argument :wink:), I can live without one, assuming there's been some manual validation.

@halter73 @martincostello If you happen to know how to add test coverage for this, feel free to jump in.

**Comment by simon-curtis** [comment]
Im just trying to load the solution up in Rider, and oh boy it's taking a while.

Is it just the `certificate.GetNameInfo(..., false)` behaivour we need to test rather then the whole method inside `CertificateLoader.LoadFromStoreCert`?

**Review by simon-curtis — COMMENTED** [review]
Change flag to named argument

**Comment by Tratcher** [comment]
@amcasey can you do a manual test for this change?

**Comment by amcasey** [comment]
> Im just trying to load the solution up in Rider, and oh boy it's taking a while.

Don't forget about Codespaces. :wink:

**Comment by amcasey** [comment]
Sorry, it took me a little while to remember how to create certs with just the right properties.  LGTM

**Review by amcasey — APPROVED** [review]

**Comment by amcasey** [comment]
Any remaining concerns @Tratcher

**Comment by dan-olsen** [comment]
@Tratcher Anyway this could be back ported to .NET 8? We only use LTS versions and I would not like to have to continue to have the workaround @simon-curtis mentioned in #49062 until .NET 10.

**Comment by Tratcher** [comment]
@dan-olsen it looks like the regression happened in 6.0 (https://github.com/dotnet/aspnetcore/pull/34582). Given the availability of a workaround, and so few reports of the issue, it's unlikely to qualify for an 8.0 patch. If this is a significant blocker for you, please open a new issue and explain why.

**Pull request merged**